proto/connector/v1/connector.proto

```diff
@@ -8,220 +8,236 @@ import "google/protobuf/struct.proto";
-        }
+    message Response {}
+  }
+
+  message Run {
+    message Request {
+      // This is the position of the record that was successfully
+      // processed.
+      bytes ack_position = 1;
     }
-
-    message Stop {
-        message Request {}
-        message Response {}
+    message Response {
+      // Record contains the record read by the source from the 3rd party
+      // resource.
+      Record record = 1;
     }
-
-    message Teardown {
-        message Request {}
-        message Response {}
+  }
+
+  message Stop {
+    message Request {}
+    message Response {
+      // This is the position of the last record in the stream, Conduit
+      // won't process records after this position anymore. After the
+      // record with this position is received by Conduit and all
+      // outstanding acknowledgments were delivered to the connector, the
+      // stream will be closed.
+      bytes last_position = 1;
     }
+  }
+
+  message Teardown {
+    message Request {}
+    message Response {}
+  }
 }
 
 message Destination {
-    message Configure {
-        message Request {
-            // Config contains the raw plugin settings.
-            map<string, string> config = 1;
-        }
-        message Response {}
+  message Configure {
+    message Request {
+      // Config contains the raw plugin settings.
+      map<string, string> config = 1;
     }
-
-    message Start {
-        message Request {}
-        message Response {}
+    message Response {}
+  }
+
+  message Start {
+    message Request {}
+    message Response {}
+  }
+
+  message Run {
+    message Request {
+      // Record contains the record that should be written to the 3rd
+      // party resource.
+      Record record = 1;
     }
-
-    message Run {
-        message Request {
-            // Record contains the record that should be written to the 3rd
-            // party resource.
-            Record record = 1;
-        }
-        message Response {
-            // This is the position of the record that was processed.
-            bytes ack_position = 1;
-            // Error should be empty if the record was successfully processed or
-            // should contain a descriptive message in case the record
-            // processing failed.
-            string error = 2;
-        }
+    message Response {
+      // This is the position of the record that was processed.
+      bytes ack_position = 1;
+      // Error should be empty if the record was successfully processed or
+      // should contain a descriptive message in case the record
+      // processing failed.
+      string error = 2;
     }
-
-    message Stop {
-        message Request {}
-        message Response {}
+  }
+
+  message Stop {
+    message Request {
+      // This is the position of the last record that was sent into the stream.
+      // Conduit won't send any records after a record with this position is
+      // received.
+      bytes last_position = 1;
```

Comment:
This field was added, the stop request contains the position of the last record written to the destination.